src/libexpr/flake/flakeref.cc

```diff
@@ -120,36 +120,38 @@ std::pair<FlakeRef, std::string> parseFlakeRefWithFragment(
                repo (i.e. the directory containing .git). The flake.nix
                seen earliest is used. */
 
-            auto flakeInfo = findFlakeDirs(absPath(path, baseDir, true), path);
-            if (flakeInfo.first) {
-                // the flake is contained in a git repo
+            auto [gitRepo, dir] = path == "." ?
+                  findFlakeDirs(absPath(path, baseDir, true), path)
+                : std::make_pair(std::nullopt, absPath(path, baseDir, true));
```

Comment:
I think this will mean that `nix build /some/path` will never treat `/some/path` as a git repository, which is problematic.
For example running `nix build $PWD` from the Nix repository will start by taking forever copying the whole directory (including the several-hundred-MBs of locally built artifacts) to the store